plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,168 @@
+    type: str
+  avatar_url:
+    description:
+      - Override the default avatar of the webhook
+    type: str
+  tts:
+    description:
+      - Set this to C(true) if this is a TTS (Test to Speech) message.
+    type: bool
+    default: false
+  embeds:
+    description:
+      - Send messages as Embeds to the discord channel
+      - Embeds can have a colored border, embedded images, text fields and more.
+    type: list
+    elements: str
+'''
+
+EXAMPLES = """
+- name: Send a message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+
+- name: Send a message to the discord channel with specific username and avatar
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+    username: Ansible
+    avatar_url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+
+- name: Send a embedded message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    embeds:
+      - title: "Embedded message"
+        description: "This is an embedded message"
+        footer:
+          text: "Author: Ansible"
+        image:
+          url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+"""
+
+RETURN = """
+http_code:
+  description:
+    - Response Code returned by discord API.
+  returned: always
+  type: int
+  sample: 204
+"""
+
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+
+
+def discord_text_msg(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+    content = module.params['content']
+    user = module.params['username']
+    avatar_url = module.params['avatar_url']
+    tts = module.params['tts']
+    embeds = module.params['embeds']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    payload = {
+        'content': content,
+        'username': user,
+        'avatar_url': avatar_url,
+        'tts': tts,
+        'embeds': embeds,
+    }
+
+    payload = module.jsonify(payload)
+
+    response, info = fetch_url(module, url, data=payload, headers=headers)
+    return response, info
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook_id=dict(type='str', required=True),
+            webhook_token=dict(type='str', required=True, no_log=True),
+            content=dict(type='str'),
+            username=dict(type='str'),
+            avatar_url=dict(type='str'),
+            tts=dict(type='bool', default=False),
+            embeds=dict(type='list', elements='str'),
+        ),
+        required_one_of=[['content', 'embeds']],
+        supports_check_mode=True
```

Comment:
If you're not going to add `check_mode`, please change this value to `False`

Reply:
Changed this. The webhook don't support any test possibility. Therefore we can't check something without sending the webhook to discords API.